pkg/database/database.go

```diff
@@ -229,6 +229,13 @@ func checkAndMigrateDB(db *DbInstance, migrate bool) error {
 			return fmt.Errorf("unsupported database schema versions alembic %s, FastTrackML %s", alembicVersion.Version, schemaVersion.Version)
 		}
 
+		runWithoutForeignKeyIfPossible := func(fn func() error) error { return fn() }
```

Comment:
this function doesn't seem to do anything except wrap another function?

Reply:
yes, by default it doesn't do anything — it only does something on SQLite

Reply:
maybe the name is confusing to me -- perhaps "disableForeignKeysFunc"? I assume for PG this is possible, just not needed?

Reply:
I called it like that because the function with use with the SQLite migrator is called `RunWithoutForeignKey`

Reply:
it's not necessary for Postgres, as generally speaking we do _not_ want to disable foreign key constraints, even whilst migrating, but SQLite's minimal support of `ALTER TABLE` forces us to do that

Reply:
"...IfNeeded"?

Reply:
addressed in https://github.com/G-Research/fasttrackml/pull/211/commits/f31805448f24abf095ee0b4dc6c8cbae55757c8f